src/ecdsa/keys.py

```diff
@@ -904,16 +913,31 @@ def from_der(cls, string, hashfunc=sha1):
 
         The only format supported for the `parameters` field is the named
         curve method. Explicit encoding of curve parameters is not supported.
-
-        While `parameters` field is defined as optional, this implementation
-        requires its presence for correct parsing of the keys.
+        In the legacy sslay format, this implementation requires the optional
+        `parameters` field to get the curve name.
+
+        The PKCS #8 format includes this object as the `privateKey` field
+        within a larger structure:
+
+            OneAsymmetricKey ::= SEQUENCE {
+                version                   Version,
+                privateKeyAlgorithm       PrivateKeyAlgorithmIdentifier,
+                privateKey                PrivateKey,
+                attributes            [0] Attributes OPTIONAL,
+                ...,
+                [[2: publicKey        [1] PublicKey OPTIONAL ]],
+                ...
+            }
 
         `publicKey` field is ignored completely (errors, if any, in it will
         be undetected).
 
         :param string: binary string with DER-encoded private ECDSA key
         :type string: bytes like object
 
+        :param pkcs8: whether to expect the data in PKCS #8 format
+        :type pkcs8: boolean
+
```

Comment:
why force the user to tell use what's the format of the file when we can detect it?

I mean, for most users the different private kye formats are completely arcane, I'd say we should be lenient here, at least by default, in what we accept.

there could be an optional parameter that forces a particular format and aborts reads if the file doesn't match, but that's more of an expert option, with limited usability.

Reply:
How do you suggest detecting the format, given only DER-encoded data? Both formats start with an integer; if it's `0` we know it's PKCS#1 but if it's `1` it could be either. Should it just detect whether the next thing after the version is an octet-string (PKCS#1) or a sequence (PKCS#8)? That feels brittle to me, but may be good enough.

Reply:
yes, detection of format of ASN.1 objects works by checking which description it matches, so looking for a sequence instead of octet-string is the correct approach